### ISSUE TITLE
Spark SQL DESCRIBE not showing proper schema on a branch

### ISSUE THREAD

**Issue by cccs-eric** [apache/iceberg#9026]
### Apache Iceberg version

1.3.1

### Query engine

Spark

### Please describe the bug 🐞

This started with a [question](https://apache-iceberg.slack.com/archives/C025PH0G1D4/p1699543073213439) on Slack and it was mentioned that this could be a bug with `DESCRIBE` not being "branch-aware".

The use-case is I have a table where I'd like to use the [W-A-P workflow](https://apache-iceberg.slack.com/archives/C025PH0G1D4/p1699543073213439), but instead of doing a simple `INSERT INTO`, I'd like to add a new column.  Here are the SQL statements used:
```SQL
CREATE TABLE my_namespace.wap_table
    using iceberg
    tblproperties ('write.parquet.compression-codec' = 'zstd',
                   'write.wap.enabled'='true'
    )
      AS
        SELECT uuid() AS table_id,
               "First row" AS value

ALTER TABLE my_namespace.wap_table CREATE branch wap_branch
ALTER TABLE my_namespace.wap_table SET TBLPROPERTIES ('write.wap.enabled'='true')

SET spark.wap.branch = wap_branch

ALTER TABLE my_namespace.wap_table ADD COLUMN nop string
```

From my understanding, the new column `nop` should only be visible from the `wab_branch` branch and not from `main`.

But both blocks of SQL will show the `nop` column:
```SQL
SET spark.wap.branch = wap_branch
DESCRIBE my_namespace.wap_table
```
```SQL
SET spark.wap.branch = main
DESCRIBE my_namespace.wap_table
```
both will return:  
| col_name	| data_type	| comment
|----------------|----------------|-------------
|table_id	|string	|None
|value	|string	|None
|nop	|string	|None

Is there a problem with my syntax (issuing DDL statements on a branch), or is there a problem with the Spark implementation?

**Comment by nastra** [comment]
@cccs-eric I took another look at this. When describing the schema of a branch, then that output should always show the schema of the current table itself and not the schema of the current snapshot. See also https://github.com/apache/iceberg/pull/9131 that addresses this behavior.

**Comment by MehulBatra** [comment]
@nastra So is this ticket closed as #9131 states it's an expected behavior, or do we need to work on this?

**Comment by nastra** [comment]
@MehulBatra yes, this is expected behavior, but sorry that I forgot to close this issue out.
